src/components/Carousel.js

```diff
@@ -79,8 +79,12 @@ class Carousel extends Component {
             selectedItem: props.selectedItem,
             hasMount: false,
             isMouseEntered: false,
-            autoPlay: props.autoPlay
+            autoPlay: props.autoPlay,
+            loop: false,
+            swipeLoop: false
         };
+
+        this.childrenLength = Children.count(this.props.children);
```

Comment:
Good point. I'll update`childrenLength` in `cDU`.

Reply:
cDU occurs after render - would need to be within cWU but that's gonna be deprecated in favour of gDS soon so might be worth just creating a function that gets the children count, or restore the original Children.count() calls. It's not really an overhead.

Reply:
I can see the new addition for cDU. It's somewhat problematic though as it will only update the children count if you go from zero children to some children. If you go from 2 children to 3 children it won't update. Also as mentioned it's after render() so it will have already rendered - ideally this needs to be in cWRP - but my preference is to restore Children.count() as it'll make things easier to read here and it's an unrelated change.

Reply:
Agreed. I will revert my change and maybe open another one specifically for this change later.